### PR TITLE
Use corlib instead of System.Private.CoreLib as a name prefix in AOT symbol names.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

**Comment by MichalStrehovsky** [comment]
Would this conflict if someone e.g. installs the [corlib](https://www.nuget.org/packages/Corlib) NuGet package that has an assembly named corlib in it?

I'm just asking because before we named the .NET Core core library System.Private.CoreLib, we briefly named it CoreLib, but there was a very popular app in the Windows Store with a CoreLib.dll assembly so we had to do a rename to something less likely to conflict.

**Comment by akoeplinger** [comment]
Is the idea to shorten the symbol names for size reasons?

**Comment by vargaz** [comment]
Updated to work around an assembly named 'corlib'.

**Comment by CoffeeFlux** [comment]
What is the exact motivation here? Size reduction?

**Comment by vargaz** [comment]
Size reduction, easier to look at assembly, etc.

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
I assume this doesn't need an AOT format version bump?

**Comment by srxqds** [comment]
> Updated to work around an assembly named 'corlib'.

it would make the assembly corlib and __corlib__ conflict?

File: src/mono/mono/mini/aot-compiler.c [hunk 1]
```diff
@@ -9564,6 +9564,17 @@ append_mangled_klass (GString *s, MonoClass *klass)
...
+static const char*
+get_assembly_prefix (MonoImage *image)
+{
+	if (mono_is_corlib_image (image))
+		return "corlib";
+	else if (!strcmp (image->assembly->aname.name, "corlib"))
+		return "__corlib__";
+	else
+		return image->assembly->aname.name;
+}
```
Comment (marek-safar) [inline src/mono/mono/mini/aot-compiler.c:9576]: Why corlib only? Cannot we map the names into some indexes for all assemblies?

Comment (vargaz) [reply src/mono/mono/mini/aot-compiler.c:9576]: This is just to make corlib llvm assembly more readable.

**Pull request merged**